proxy/http2/unit_tests/test_HpackIndexingTable.cc

```diff
@@ -71,7 +71,10 @@ TEST_CASE("HPACK low level APIs", "[hpack]")
       HpackIndexingTable indexing_table(4096);
 
       for (const auto &i : indexed_test_case) {
-        std::unique_ptr<HTTPHdr> headers(new HTTPHdr);
+        std::unique_ptr<HTTPHdr, void (*)(HTTPHdr *)> headers(new HTTPHdr, [](HTTPHdr *hdr) {
+          hdr->destroy();
+          delete hdr;
```

Comment:
This gets reused a few times in here. Can we:
1. Create a separate function for this.
2. Add a comment there explaining why this is destroy call is needed. (Basically `HTTPHdr` needs to be destroyed as well as deleted to be cleaned up properly).

Reply:
Done. Please take another look.